src/main/java/com/google/cloud/teleport/templates/PubsubToText.java

```diff
@@ -165,10 +162,9 @@ public static PipelineResult run(Options options) {
                         options.getOutputFilenamePrefix(),
                         options.getOutputShardTemplate(),
                         options.getOutputFilenameSuffix()))
-                .withTempDirectory(NestedValueProvider.of(
-                    options.getOutputDirectory(),
-                    (SerializableFunction<String, ResourceId>) input ->
-                        FileBasedSink.convertToFileResourceIfPossible(input))));
+                .withTempDirectory(
+                    FileBasedSink.convertToFileResourceIfPossible(options.getTempLocation())
+                        .getCurrentDirectory()));
```

Comment:
Without the NestedValueProvider, the tempLocation specified at the time this template is built and released (compile time) will be used at runtime regardless of the used provided value of tempLocation. This would not work since the users of the public template in Dataflow do not have access to write to the tempLocation specified at compile time.

IMO, there are two options here:
1. Create a custom template with an additional option (--userTempLocation perhaps?) that is a ValueProvider and would be used instead of the OutputDirectory. This would be non backward compatible change for us
2. Take a look at [DualInputNestedValueProvider](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/645836749a2960347ec7285f112f0b19a477fb3a/src/main/java/com/google/cloud/teleport/util/DualInputNestedValueProvider.java#L29) and how it was used in other templates. IMO, this can be used to provide a backward compatible optional flag (if userTempLocation is provided then use that OR else use outputDirectory).


1.

Reply:
Thank you for feedback. I went for option 2. Please see updated code.